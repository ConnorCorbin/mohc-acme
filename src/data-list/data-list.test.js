import React from 'react';

import DataList from 'data-list/data-list';

describe('DataList', () => {
  let wrapper;
  let expectedValues;
  let propValues;
  const getWrapper = ({
    listName = 'Data-list-name',
    dataListOptions = ['option1', 'option2', 'option3'],
  } = {}) => shallow(
    <DataList
    listName={listName}
    dataListOptions={dataListOptions}
    />,
  );

  it('should render component', () => {
    wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toEqual(false);
  });

  it('should render component with correct ID for datalist HTML element', () => {
    wrapper = getWrapper();

    expect(wrapper.find('datalist').props().id).toEqual('Data-list-name');
  });

  it('should render component with correct option values', () => {
    wrapper = getWrapper();

    expectedValues = ['option1', 'option2', 'option3'];
    propValues = wrapper.find('option').map(value => value.props().value);
    expect(propValues).toEqual(expectedValues);
  });

  it('should render component with correct option key values', () => {
    wrapper = getWrapper();

    expectedValues = ['option10', 'option21', 'option32'];
    propValues = wrapper.find('option').map(value => value.key());
    expect(propValues).toEqual(expectedValues);
  });
});
