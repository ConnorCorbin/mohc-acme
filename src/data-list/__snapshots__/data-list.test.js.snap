// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataList should render component 1`] = `
<datalist
  id="Data-list-name"
>
  <option
    key="option10"
    value="option1"
  />
  <option
    key="option21"
    value="option2"
  />
  <option
    key="option32"
    value="option3"
  />
</datalist>
`;
